src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/RawConnectionStream.cs

```diff
@@ -74,7 +74,7 @@ public override async ValueTask<int> ReadAsync(Memory<byte> buffer, Cancellation
                     }
                 }
 
-                if (bytesRead == 0)
+                if (bytesRead == 0 && buffer.Length != 0)
```

Comment:
Does this deserve a comment?

```suggestion
                // Don't close the connection for a zero-byte read.
                if (bytesRead == 0 && buffer.Length != 0)
```

Reply:
Wouldn't hurt :)

I'll add these into the main branch tho, no point in doing so just in the backport.